### PR TITLE
SDI-129: ✨ Add in call to absence reasons v2

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: backend/controllers/attendance/attendance.ts [hunk 1]
```diff
@@ -62,8 +62,12 @@ export const attendanceFactory = (whereaboutsApi) => {
   }
 
   const getAbsenceReasons = async (context) => {
-    const absenceReasons = await whereaboutsApi.getAbsenceReasons(context)
+    const [absenceReasons, absenceReasonsV2] = await Promise.all([
+      await whereaboutsApi.getAbsenceReasons(context),
+      await whereaboutsApi.getAbsenceReasonsV2(context),
```
Comment (petergphillips) [inline backend/controllers/attendance/attendance.ts:67]: we call both whereabouts api absence reasons endpoints for now and then combine results

Comment (hyderkhanmoj) [reply backend/controllers/attendance/attendance.ts:67]: Is the idea to remove the v1 version in the future?

**Review by hyderkhanmoj — APPROVED** [review]
LGTM

File: backend/controllers/attendance/attendance.ts (hunk 1 above)
Comment (petergphillips) [reply backend/controllers/attendance/attendance.ts:67]: Yes, well sort of, it is called from multiple places within DPS so having to do a bit at a time.  Idea is to gradually migrate all to new call, then copy `/v2` to the existing (now unused) endpoint, move everything back to the existing endpoint and then remove `/v2`.

**Pull request merged**